standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1229,74 +1229,78 @@ private void create_database_core(RawStore ms, final Database db)
     Map<String, String> transactionalListenersResponses = Collections.emptyMap();
     try {
       firePreEvent(new PreCreateDatabaseEvent(db, this));
-      //reinstate location uri for metastore db.
-      if (skipAuthorization == true){
```

Comment:
could we extract the creating directory logic to a standalone method, so it's clear? just a suggestion.

Reply:
We can leave it the same way this time. Because this change is not highly related with the HMS directory skipping.